src/fs/filesystem.ts

```diff
@@ -145,24 +132,14 @@ export class FileSystem {
     })
   }
 
-  async pinList(): Promise<CID[]> {
-    const privateResult = await this.privateTree.putWithPins()
-    const publicResult = await this.publicTree.putWithPins()
-    const rootCID = await this.sync()
-    return [
-      ...privateResult.pins,
-      ...publicResult.pins,
-      rootCID
-    ]
+  async updatePinTree(pins: PinMap): Promise<void> {
+    const pinLinks = await pinMapToLinks('private', pins) // @@TODO: add salt
```

Comment:
Still need to add salt, but first the Filesystem needs to be aware of a user's DID. Which means that we probably can't just instantiate it from a `cid`...

Reply:
ok added salt
Now if you instantiate an FFS `fromUsername`, it also stores your DID which it can use for a salt. If you just instantiate an FFS `fromCID` or `empty` then takes an optional param `rootDID` and defaults to an empty string

![](https://media1.giphy.com/media/3o7P4F86TAI9Kz7XYk/giphy.gif)